File: frontend/lib/types/course-like.ts
/**
 * Course Reaction types (YouTube-style like/dislike)
 */

export type ReactionType = 'like' | 'dislike';

export interface CourseReactionStatus {
  user_reaction: ReactionType | null;
  like_count: number;
  dislike_count: number;
}

export interface CourseReactionToggleResponse {
  user_reaction: ReactionType | null;
  like_count: number;
  dislike_count: number;
  message: string;
}

// Keep aliases for backward compatibility
export type CourseLikeStatus = CourseReactionStatus;
export type CourseLikeToggleResponse = CourseReactionToggleResponse;

File: backend/app/schemas/course_like.py
"""
Course Reaction schemas for request/response validation (YouTube-style like/dislike)
"""
from typing import Optional, Literal
from pydantic import BaseModel


class CourseReactionRequest(BaseModel):
    """Request schema for toggling reaction"""
    reaction_type: Literal["like", "dislike"]


class CourseReactionResponse(BaseModel):
    """Response schema for course reaction status"""
    user_reaction: Optional[Literal["like", "dislike"]] = None
    like_count: int
    dislike_count: int


class CourseReactionToggleResponse(BaseModel):
    """Response schema after toggling reaction"""
    user_reaction: Optional[Literal["like", "dislike"]] = None
    like_count: int
    dislike_count: int
    message: str


# Keep aliases for backward compatibility
CourseLikeResponse = CourseReactionResponse
CourseLikeToggleResponse = CourseReactionToggleResponse
